utils/format.go

```diff
@@ -332,6 +332,30 @@ func FormatBlockStatusShort(status uint64) template.HTML {
 	}
 }
 
+func FormatBlockStatusStyle(status uint64) template.HTML {
+	if status == 0 {
+		return `<div title="This slot is still scheduled" data-toggle="tooltip" class="style-badge style-badge-single-char style-bg-neutral-1"><p class="style-status-tag-text" >S<span class="d-none d-sm-inline">cheduled</span></p></div>`
+	} else if status == 1 {
+		return `<div title="This block has been proposed" data-toggle="tooltip" class="style-badge style-badge-single-char style-bg-good"><p class="style-status-tag-text text-white" >P<span class="d-none d-sm-inline">roposed</span></p></div>`
+	} else if status == 2 {
+		return `<div title="This block has been proposed" data-toggle="tooltip" class="style-badge style-badge-single-char style-bg-bad"><p class="style-status-tag-text" >M<span class="d-none d-sm-inline">issed</span></p></div>`
+	} else if status == 3 {
+		return `<div title="This block has been orphaned" data-toggle="tooltip" class="style-badge style-badge-single-char style-bg-neutral-2"><p class="style-status-tag-text text-white" >O<span class="d-none d-sm-inline">rphaned</span></p></div>`
+	} else {
+		return `<div title="This shouldn't be possible" data-toggle="tooltip" class="style-badge style-bg-neutral-2"><p class="style-status-tag-text text-white" >?</p></div>`
+	}
+}
+
+func FormatEpochStatus(finalized bool, participationRate float64) template.HTML {
+	if finalized {
+		return `<div title="This epoch is finalized" data-toggle="tooltip" class="style-badge style-badge-single-char style-bg-good"><p class="style-status-tag-text text-white" >F<span class="d-none d-sm-inline">inalized</span></p></div>`
+	} else if participationRate > 0.66 && participationRate < 1 {
```

Comment:
Personally I think this `< 1` is confusing as it excludes 1. Why would a perfect participation rate of 1 not be considered save?
Epochs with a participationRate of exactly 1, to my knowledge, are more than rare but sill. Since this is a workaround for the newest epoch in the db, I suggest switching this to `<= 1` as the newest epoch is not finalized anyways.

This applies to line 648 in services.go (https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1718/files#diff-3b37e5382c8ce5b2244a5256b612ea8e965943bc02ec783669c2e679e69d6760R648). A check for finalized should suffice, should it not?

And I know, this is very minor yet the page would show a (more than unlikely) perfect epoch as not finalized.

Reply:
The participation rate for the most recent epoch in the database is always 1, which is why I check for < 1 instead of <= 1. Checking for <= 1 would incorrectly mark the most recent epoch as safe. I realize that this is a convoluted workaround and I agree that it's not ideal.
Regarding line 648, I believe that the EpochParticipationRate field in the data object represents the participation rate of the most recent safe epoch, which does not necessarily entail said epoch being finalized. I also noticed that the first condition of the if statement is unnecessary, so I plan to remove it.

Reply:
Got it. Thanks for adding the comments => Resolved